### DISCUSSION TITLE
Lazy loading JavaScript FilamentAsset::getScriptSrc reports an error

### DISCUSSION THREAD

**Discussion by killjin** [filamentphp/filament#7275] [Help]
see doc: https://beta.filamentphp.com/docs/3.x/support/assets

```php
if (app()->runningInConsole()) {
    FilamentAsset::register([
        Js::make('custom-script', __DIR__ . '/../../resources/js/custom.js'),
    ]);
}

// xxx.blade.php
<div x-load-js="[@js(\Filament\Support\Facades\FilamentAsset::getScriptSrc('custom-script'))]"></div>
```
In this way, there is no actual registration, and an error will be reported:
throw new Exception("Script with ID [{$id}] not found for package [{$package}].");

FilamentAsset::getScriptSrc should directly return the published address according to the package name and id, instead of getting it from the registered $script.

x-load-js is executed after x-init, and some code cannot be executed after asynchronously loading framework js. We are writing a plugin, and we want to load jquery only on the plugin page, and execute our custom code after jquery is loaded or in x-init, how to do it?

@push('scripts') is a good way, but it doesn't work now. 
The solution of Filament::registerRenderHook is not good. I can only write js code in php. How to implement it in blade?

**Comment by danharrin** [comment]
The docs are wrong, runningInConsole is not required anymore. You should use this

```
FilamentAsset::register([
    Js::make('custom-script', __DIR__ . '/../../resources/js/custom.js')->loadedOnRequest(),
]);
```
